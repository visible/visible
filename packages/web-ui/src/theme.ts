export interface Theme {
  foreground: {
    normal: string;
    wash: string;
  };
  background: {
    normal: string;
    wash: string;
  };
  highlight: {
    normal: string;
  };
  border: {
    normal: string;
  };
}
