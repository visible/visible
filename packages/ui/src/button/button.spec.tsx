import '@testing-library/jest-dom/extend-expect';

import { render } from '@testing-library/react';
import React from 'react';
import { ThemeProvider } from 'styled-components';

import { theme } from '../theme';
import { Button } from '.';

describe('Button', () => {
  it('renders primary button', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Button appearance="primary" />
      </ThemeProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders skeleton button', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Button appearance="skeleton" />
      </ThemeProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
