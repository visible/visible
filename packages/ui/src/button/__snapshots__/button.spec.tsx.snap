// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders primary button 1`] = `
.c1 {
  box-sizing: border-box;
  padding: 8px 14px;
  -webkit-transition: 0.15s ease-in;
  transition: 0.15s ease-in;
  border: none;
  border-radius: 4px;
  font-size: 14px;
  cursor: pointer;
  background-color: #ffaa01;
  box-shadow: 0 1.5px 3px rgba(255,170,1,0.2);
  color: white;
}

.c1:hover.c0:not(:disabled) {
  -webkit-transition: 0.15s ease-out;
  transition: 0.15s ease-out;
  opacity: 0.8;
}

.c1:focus {
  -webkit-transition: 0.15s ease-out;
  transition: 0.15s ease-out;
  outline: none;
  box-shadow: 0 0 0 3px rgba(255,170,1,0.4);
}

.c1:active {
  opacity: 1 !important;
}

.c1:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

<button
  class="c0 c1"
/>
`;

exports[`Button renders skeleton button 1`] = `
.c1 {
  box-sizing: border-box;
  padding: 8px 14px;
  -webkit-transition: 0.15s ease-in;
  transition: 0.15s ease-in;
  border: none;
  border-radius: 4px;
  font-size: 14px;
  cursor: pointer;
  border: 1px solid #ffaa01;
  color: #ffaa01;
}

.c1:hover.c0:not(:disabled) {
  -webkit-transition: 0.15s ease-out;
  transition: 0.15s ease-out;
  opacity: 0.8;
}

.c1:focus {
  -webkit-transition: 0.15s ease-out;
  transition: 0.15s ease-out;
  outline: none;
  box-shadow: 0 0 0 3px rgba(255,170,1,0.4);
}

.c1:active {
  opacity: 1 !important;
}

.c1:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

<button
  class="c0 c1"
/>
`;
